source/dub/dub.d

```diff
@@ -172,19 +162,21 @@ class Dub {
 	private void init()
 	{
 		import std.file : tempDir;
-		version(Windows){
-			m_systemDubPath = Path(environment.get("ProgramData")) ~ "dub/";
-			m_userDubPath = Path(environment.get("APPDATA")) ~ "dub/";
+		version(Windows) {
+			m_dirs.systemSettings = Path(environment.get("ProgramData")) ~ "dub/";
+			m_dirs.userSettings = Path(environment.get("APPDATA")) ~ "dub/";
 		} else version(Posix){
-			m_systemDubPath = Path("/var/lib/dub/");
-			m_userDubPath = Path(environment.get("HOME")) ~ ".dub/";
-			if(!m_userDubPath.absolute)
-				m_userDubPath = Path(getcwd()) ~ m_userDubPath;
+			m_dirs.systemSettings = Path("/var/lib/dub/");
+			m_dirs.userSettings = Path(environment.get("HOME")) ~ ".dub/";
+			if (!m_dirs.userSettings.absolute)
+				m_dirs.userSettings = Path(getcwd()) ~ m_dirs.userSettings;
 		}
 
-		m_tempPath = Path(tempDir);
-		m_userConfig = jsonFromFile(m_userDubPath ~ "settings.json", true);
-		m_systemConfig = jsonFromFile(m_systemDubPath ~ "settings.json", true);
+		m_dirs.temp = Path(tempDir);
+
+		m_config = new DubConfig(jsonFromFile(m_dirs.userSettings ~ "settings.json", true), null);
+		m_config = new DubConfig(jsonFromFile(m_dirs.systemSettings ~ "settings.json", true), m_config);
+		m_config = new DubConfig(jsonFromFile(Path(thisExePath) ~ "../etc/dub/settings.json", true), m_config);
```

Comment:
If I understand correctly, this ordering means that first `../etc/dub/settings.json` is used, then `/etc/`, and then `~/.dub/`?